pkg/operation/care/checker.go

```diff
@@ -229,31 +229,47 @@ func (b *HealthChecker) checkNodes(condition gardencorev1beta1.Condition, nodes
 
 // CheckManagedResource checks the conditions of the given managed resource and reflects the state in the returned condition.
 func (b *HealthChecker) CheckManagedResource(condition gardencorev1beta1.Condition, mr *resourcesv1alpha1.ManagedResource) *gardencorev1beta1.Condition {
+	conditionsToCheck := map[gardencorev1beta1.ConditionType]func(status gardencorev1beta1.ConditionStatus) bool{
+		resourcesv1alpha1.ResourcesApplied: defaultSuccessfulCheck(),
+		resourcesv1alpha1.ResourcesHealthy: defaultSuccessfulCheck(),
+	}
+
+	return b.checkManagedResourceConditions(condition, mr, conditionsToCheck)
+}
+
+func defaultSuccessfulCheck() func(status gardencorev1beta1.ConditionStatus) bool {
+	return func(status gardencorev1beta1.ConditionStatus) bool {
+		return status != gardencorev1beta1.ConditionFalse && status != gardencorev1beta1.ConditionUnknown
```

Comment:
```suggestion
		return status == gardencorev1beta1.ConditionTrue
```

? 😄 In the light of this, do we still need this function in the first place?

Reply:
With @ialidzhikov we were also discussing  this and wondered whether to consider `Progressing` status as successful as well (as it was previously done). As far as I see the only condition that can have `Progressing` status currently is `ResourcesApplied` but I wasn't sure what to expect in the future.
If we only want to consider a condition as "successful" if it is `True` then i will get rid of these functions

Reply:
Hm, if at least one `ManagedResource` has `Progressing` status then we probably also should set the `Seed` condition to `Progressing` (unless there are `ManagedResource`s with `False` status). Only when all `ManagedResource`s have `True` status the `Seed` condition should also be `True`. But maybe this is more something related to #4722 and we should skip it for now. So, I'm fine proceeding as suggested.